### PR TITLE
Projectile GUI homework

### PR DESCRIPTION
**Pull request by rbral**
Screenshot 2024-02-20 201738

### CODE REVIEW

File: src/main/java/bral/projectile/ProjectileFrame.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                        Double.parseDouble(velocityField.getText())
+                );
+                projectile.setSeconds(
+                        Double.parseDouble(secondsField.getText())
+                );
+                // display info
+                xResultLabel.setText(Double.toString(projectile.getX()));
+                yResultLabel.setText(Double.toString(projectile.getY()));
+                peakYResultLabel.setText(Double.toString(projectile.getPeakY()));
+                interceptXResultLabel.setText(Double.toString(projectile.getInterceptX()));
```
Comment (andrewoid) [inline src/main/java/bral/projectile/ProjectileFrame.java]: Isn't this recalculateAndDisplay()

File: src/main/java/bral/projectile/ProjectileFrame.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+            }
+        });
+
+    }
+
+    private void recalculateAndDisplay(
+            JSlider angleSlider, JTextField velocityField, JTextField secondsField,
+            JLabel xResultLabel, JLabel yResultLabel, JLabel peakYResultLabel, JLabel interceptXResultLabel
```
Comment (andrewoid) [inline src/main/java/bral/projectile/ProjectileFrame.java]: If you make these Components member variables then you don't need to pass them.

File: src/main/java/bral/projectile/ProjectileFrame.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            );
+            // display info
+            xResultLabel.setText(Double.toString(projectile.getX()));
+            yResultLabel.setText(Double.toString(projectile.getY()));
+            peakYResultLabel.setText(Double.toString(projectile.getPeakY()));
+            interceptXResultLabel.setText(Double.toString(projectile.getInterceptX()));
+        }
+        catch (NumberFormatException e)
+        {
+            e.getMessage();
```
Comment (andrewoid) [inline src/main/java/bral/projectile/ProjectileFrame.java]: This doesn't do anything.
What do you want to do?

File: src/main/java/bral/projectile/SimpleDocumentListener.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package bral.projectile;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+
+@FunctionalInterface
+public interface SimpleDocumentListener extends DocumentListener {
+    void update(DocumentEvent e);
```
Comment (andrewoid) [inline src/main/java/bral/projectile/SimpleDocumentListener.java]: Where did you see this?

File: src/main/java/bral/projectile/ProjectileFrame.java (hunk 3 above)
Comment (rbral) [reply src/main/java/bral/projectile/ProjectileFrame.java]: It throws an error if not all the fields are filled in yet (when the user enters the numbers for the first time)

File: src/main/java/bral/projectile/SimpleDocumentListener.java (hunk 4 above)
Comment (rbral) [reply src/main/java/bral/projectile/SimpleDocumentListener.java]: I did this from the Stackoverflow that was emailed to us as instructions.

File: src/main/java/bral/projectile/ProjectileFrame.java (hunk 3 above)
Comment (andrewoid) [reply src/main/java/bral/projectile/ProjectileFrame.java]: I mean the specific line
e getMessage()
Doesn't do anything

File: src/main/java/bral/projectile/ProjectileFrame.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                    Double.parseDouble(secondsField.getText())
+            );
+            // display info
+            resultLabelX.setText(Double.toString(projectile.getX()));
+            resultLabelY.setText(Double.toString(projectile.getY()));
+            resultLabelPeakY.setText(Double.toString(projectile.getPeakY()));
+            resultLabelInterceptX.setText(Double.toString(projectile.getInterceptX()));
+        } catch (NumberFormatException e)
+        {
+            System.out.println("Did not fill in all fields.");
```
Comment (andrewoid) [inline src/main/java/bral/projectile/ProjectileFrame.java]: e.printStackTrace();

**Review by andrewoid — APPROVED** [review]

**Pull request merged**